eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreClientSpec.scala

```diff
@@ -1374,39 +1491,195 @@ class BitcoinCoreClientSpec extends TestKitBaseClass with BitcoindService with A
+        wallet.fundTransaction(txNotFunded, FeeratePerKw(2500 sat), replaceable = true, externalWeight).pipeTo(sender.ref)
+        signTransaction(wallet, sender.expectMsgType[FundTransactionResponse].tx).pipeTo(sender.ref)
+        val partiallySignedTx = sender.expectMsgType[SignTransactionResponse].tx
+        val anchorIndex = partiallySignedTx.txIn.indexWhere(_.outPoint == OutPoint(localCommitTx, 1))
+        val sig = Transaction.signInput(partiallySignedTx, anchorIndex, Scripts.anchor(priv1.publicKey), SigHash.SIGHASH_ALL, 330 sat, SigVersion.SIGVERSION_WITNESS_V0, priv1)
+        val signedTx = partiallySignedTx.updateWitness(anchorIndex, Scripts.witnessAnchor(der2compact(sig), Script.write(Scripts.anchor(priv1.publicKey))))
+        assert(signedTx.txIn.length == 2) // only one wallet input should have been added
+        wallet.publishTransaction(signedTx).pipeTo(sender.ref)
+        sender.expectMsg(signedTx.txid)
+        signedTx
+      }
+      localAnchorTx
+    }
+    // Both wallet utxos are now spent.
+    walletUtxos.foreach(utxo => assert(spliceTx.txIn.exists(_.outPoint.txid == utxo.txid) || anchorTx.txIn.exists(_.outPoint.txid == utxo.txid)))
+
+    // At that point, the mempool contains the following transactions:
+    //                   +------------------+     +------------------+     +-----------+
+    // funding_tx_1 ---->| remote_splice_tx |---->| remote_splice_tx |---->|           |
+    //                   +------------------+     +------------------+     | splice_tx |
+    //                                                  wallet_utxo_1 ---->|           |
+    //                                                                     +-----------+
+    //                   +-----------------+     +-----------+
+    // funding_tx_2 ---->| local_commit_tx |---->|           |
+    //                   +-----------------+     | anchor_tx |
+    //                        wallet_utxo_2 ---->|           |
+    //                                           +-----------+
+
+    // We now fill the mempool with large transactions to evict splice_tx and anchor_tx.
+    // We start with very large transactions and then use smaller transaction to completely fill the mempool.
+    parentTxs.take(12).foreach(tx => publishLargeTx(tx, 0, largeInputsCount, FeeratePerKw(5_000 sat)))
+    (0 until largeInputsCount).foreach(i => publishLargeTx(parentTxs.last, i, i + 1, FeeratePerKw(5_000 sat)))
+    Seq(spliceTx, anchorTx).foreach(walletTx => {
+      wallet.getMempoolTx(walletTx.txid).pipeTo(sender.ref)
+      assert(sender.expectMsgType[Failure].cause.getMessage.contains("Transaction not in mempool"))
+    })
+
+    // The wallet transactions have been evicted, but they must be kept inside the wallet and not be double-spent.
+    val txNotFunded = Transaction(2, Nil, Seq(TxOut(50_000 sat, Script.pay2wpkh(randomKey().publicKey))), 0)
+    wallet.fundTransaction(txNotFunded, FeeratePerKw(5000 sat), replaceable = true).pipeTo(sender.ref)
```

Comment:
I wonder why you didn't use methods like `listlockunspent`, `gettxout`, or `listunspent` to check this? Or did you want to actually use `fundrawtransaction` to be as close as possible to the reality?